forms-shared/src/summary-json/SummaryXmlForm.tsx

```diff
@@ -0,0 +1,227 @@
+      ))}
+    </SummaryXmlFormTag.Array>
+  )
+}
+
+const ObjectFieldTemplate = ({ properties, idSchema, title }: ObjectFieldTemplateProps) => {
+  const id = idSchema.$id
+  const splitId = id.split('_')
+  const isFormObject = splitId.length === 1 && splitId[0] === 'root'
+  const isStepObject = splitId.length === 2 && splitId[0] === 'root'
+
+  const content = properties.map((element, index) => (
+    <Fragment key={index}>{element.content}</Fragment>
+  ))
+
+  if (isFormObject) {
+    return <SummaryXmlFormTag.Form id={id}>{content}</SummaryXmlFormTag.Form>
+  }
+
+  if (isStepObject) {
+    const stepName = splitId[1]
+    return (
+      <SummaryXmlFormTag.Step id={id} title={title} name={stepName}>
+        {content}
+      </SummaryXmlFormTag.Step>
+    )
+  }
+
+  return <>{content}</>
+}
+
+const theme: ThemeProps = {
+  templates: {
+    FieldTemplate,
+    ObjectFieldTemplate,
+    ArrayFieldTemplate,
+    ArrayFieldItemTemplate: ArrayFieldItemTemplate as ComponentType<ArrayFieldTemplateItemType>,
+  },
+  widgets: {
+    Select: wrapWidget('Select'),
+    Input: wrapWidget('Input'),
+    RadioGroup: wrapWidget('RadioGroup'),
+    TextArea: wrapWidget('TextArea'),
+    Checkbox: wrapWidget('Checkbox'),
+    CheckboxGroup: wrapWidget('CheckboxGroup'),
+    FileUpload: wrapWidget('FileUpload'),
+    DatePicker: wrapWidget('DatePicker'),
+    TimePicker: wrapWidget('TimePicker'),
+    CustomComponents: () => {
+      return null
+    },
+  } satisfies Record<BaWidgetType, ComponentType<WidgetProps>>,
+  fields: {
+    AnyOfField: (props) => {
+      // RJSF renders unnecessary <div>s for MultiSchemaField, which breaks the XML structure.
+      // https://github.com/rjsf-team/react-jsonschema-form/blob/ec932db942dd046640303056c89e3501b16ec469/packages/core/src/components/fields/MultiSchemaField.tsx#L217
+      const options = getUiOptions(props.uiSchema)
+      if (options?.widget === 'CustomComponents') {
+        return null
+      }
+
+      const defaultRegistry = getDefaultRegistry()
+      const DefaultAnyOfField = defaultRegistry.fields.AnyOfField
+
+      return <DefaultAnyOfField {...props} />
+    },
+  },
+}
+
+const ThemedForm = withTheme(theme)
+
+type SummaryXmlFormProps = Pick<FormProps, 'schema' | 'uiSchema' | 'formData'>
+
+/**
+ * Generates a summary XML form based on the provided schema, UI schema, and form data.
+ *
+ * Getting a summary for JSON Schema data is hard and is not supported by RJSF out of the box. This component leverages
+ * that RJSF allows to customize the form rendering and generates a custom XML structure that represents the form data.
+ * Unfortunately, it is not possible to generate a JSON summary directly, so the XML is later parsed into JSON.
+ * The generated XML is tightly coupled with its parsing in `getSummaryJson` function, and it is not used anywhere else.
+ */
+const SummaryXmlForm = ({ schema, uiSchema, formData }: SummaryXmlFormProps) => {
```

Comment:
So this renders "regular" HTML, right ? If so, would perhaps rename, along with the rest of the functions where this is implied ? Less because of XML/HTML purism (though partly, because we're parsing this using window.DOMParser) and more because you can force react to render custom xml and this looks like we're doing just that. And where not as far as I can tell.